src/client/datascience/context/telemetry.ts

```diff
@@ -75,13 +75,110 @@ export function getErrorClassification(error: Error) {
         return 'jupyterconnection';
     } else if (isErrorType(error, JupyterInstallError)) {
         return 'jupyterinstall';
+    } else if (isErrorType(error, PythonKernelDiedError)) {
+        return getReasonForKernelToDie(error);
     } else if (isErrorType(error, KernelDiedError)) {
         return 'kerneldied';
     } else if (isErrorType(error, FetchError)) {
         return 'fetcherror';
     }
     return 'unknown';
 }
+/**
```

Comment:
Not happy with this. We'll only ever identify problems we already know about. Which are theoretically useless once we fix them.

Reply:
Can we wait until Olivia responds? Or think about another way we can achieve uniqueness that we can track down

Reply:
Had another idea. 

We could identify the set of site-packages that we know have potential errors thrown in them. Those would be okay to send as they're not user information.

So that list might be:

```
ipykernel
jupyter
zmq
traitlets
sys
os
jupyter-client
nb-client
``` 

At least this would catch more errors.

Reply:
Will do that and leave this code as is.

Reply:
I.e. send the hash of the folder & file from the last frame

Reply:
Sounds good. We can add as many packages as we can think of to that list.

Reply:
Oops, not yet completed
I didn't take the folders from site-packages into account.

Reply:
Done.